isis/src/qisis/objs/ControlHealthMonitorWidget/ControlHealthMonitorWidget.cpp

```diff
@@ -0,0 +1,769 @@
+#include <QPieSeries>
+#include <QPieSlice>
+#include <ControlNet.h>
+#include <ControlNetVitals.h>
+#include <ControlPoint.h>
+
+
+namespace Isis {
+
+  QT_CHARTS_USE_NAMESPACE
+
+  /**
+   * This class is the front end representation of a ControlNetVitals object.
+   * It will accept a ControlNetVitals object upon initialization and reflects the current
+   * real-time status of the embedded ControlNet in the ControlNetVitals object.
+   *
+   * @param vitals (ControlNetVitals *) The ControlNetVitals object that contains the ControlNet.
+   * @param parent (QWidget *) Pointer to parent widget
+   */
+  ControlHealthMonitorWidget::ControlHealthMonitorWidget(ControlNetVitals *vitals, QWidget *parent) : QWidget(parent) {
+
+    createGui();
+    m_vitals = vitals;
+    connect (m_vitals, SIGNAL(networkChanged()),
+            this, SLOT(update()));
+    update();
+  }
+
+  /**
+   *  This SLOT is called whenever the is a change made to the network embedded in the
+   *  Global m_vitals object. Changes are detected via the networkChanged() signal which
+   *  is emitted from the ControlNetVitals object which is triggered whenever
+   *  networkStructureModified() is emitted from the embedded ControlNet.
+   *
+   */
+  void ControlHealthMonitorWidget::update() {
+    m_numImagesLabel->setText("Images:  " + toString(m_vitals->numImages()));
+    m_numPointsLabel->setText("Points " + toString(m_vitals->numPoints()));
+    m_numMeasuresLabel->setText("Measures: " + toString(m_vitals->numMeasures()));
+    m_netLabel->setText("Control Network: " + m_vitals->getNetworkId());
+    m_statusLabel->setText(m_vitals->getStatus());
+    m_statusDetails->setText(m_vitals->getStatusDetails());
+    m_imagesMeasuresValue->setText(toString(m_vitals->numImagesBelowMeasureThreshold()));
+    m_imagesHullValue->setText(toString(m_vitals->numImagesBelowHullTolerance()));
+    m_pointsIgnoredLabel->setText(toString(m_vitals->numIgnoredPoints()));
+    m_pointsFreeLabel->setText(toString(m_vitals->numFreePoints()));
+    m_pointsFixedLabel->setText(toString(m_vitals->numFixedPoints()));
+    m_pointsConstrainedLabel->setText(toString(m_vitals->numConstrainedPoints()));
+    m_pointsEditLockedLabel->setText(toString(m_vitals->numLockedPoints()));
+    m_pointsFewMeasuresLabel->setText(toString(m_vitals->numPointsBelowMeasureThreshold()));
+
+    // We should enumerate the network state and do a comparison on enums here, not strings.
+    if (m_vitals->getStatus() == "Broken!") updateStatus(0);
+    else if (m_vitals->getStatus() == "Weak!") updateStatus(1);
+    else if (m_vitals->getStatus() == "Healthy!") updateStatus(2);
+
+    QPieSeries *series = new QPieSeries();
+    series->append("Ignored", m_vitals->numIgnoredPoints());
+    series->append("Locked", m_vitals->numLockedPoints());
+    series->append("Free", m_vitals->numFreePoints());
+    series->append("Constrained", m_vitals->numConstrainedPoints());
+    series->append("Fixed", m_vitals->numFixedPoints());
```

Comment:
these 5 things are not mutually exclusive. Ignored and locked are separate boolean qualities of the points. The Free/Constrained/Fixed properties are mutually exclusive.